pkgs/build-support/setup-hooks/auto-patchelf.sh

```diff
@@ -4,23 +4,22 @@ gatherLibraries() {
     autoPatchelfLibs+=("$1/lib")
 }
 
+# This error is ignored, because $targetOffset is not defined in this script.
+# shellcheck disable=SC2154
 addEnvHooks "$targetOffset" gatherLibraries
 
 isExecutable() {
     [ "$(file -b -N --mime-type "$1")" = application/x-executable ]
 }
 
 findElfs() {
-    find "$1" -type f -exec "$SHELL" -c '
-        while [ -n "$1" ]; do
-            mimeType="$(file -b -N --mime-type "$1")"
-            if [ "$mimeType" = application/x-executable \
-              -o "$mimeType" = application/x-sharedlib ]; then
-                echo "$1"
-            fi
-            shift
-        done
-    ' -- {} +
+    while IFS='' read -r -d $'\0' file; do
+        mimeType="$(file -b -N --mime-type "$file")"
+        if [ "$mimeType" = application/x-executable ] || \
+            [ "$mimeType" = application/x-sharedlib ]; then
+            echo "$file"
+        fi
+    done < <(find "$1" -type f -print0)
```

Comment:
I don't think this is a big improvement over the old implementation, because IMHO it makes the code less readable. What exactly was the problem with `--`? Was it something that `shellcheck` reported or did the error occur at runtime?